### PR TITLE
Switch satisfied by arguments

### PR DESCRIPTION
**Pull request by alcohol**
Gonne give you some small contained PR's @Seldaek  :p

### CODE REVIEW

File: src/Semver.php [hunk 1]
```diff
@@ -45,23 +45,15 @@ public static function satisfies($version, $constraints)
...
-        }
-
-        $versionParser = self::$versionParser;
-        $constraints = $versionParser->parseConstraints($constraints);
-        $versions = array_filter($versions, function ($version) use ($constraints, $versionParser) {
-            $version = new Constraint('==', $versionParser->normalize($version));
-
-            return $constraints->matches($version);
+        $versions = array_filter($versions, function ($version) use ($constraints) {
+            return self::satisfies($version, $constraints);
```
Comment (Seldaek) [inline src/Semver.php]: Needs `$that = $this;` and `use ($that)` and `$that->satisfies()` to pass on 5.3 :/

Comment (alcohol) [inline src/Semver.php]: I was looking at that, but it's a static so that's not going to work.

Comment (Seldaek) [inline src/Semver.php]: It will, you can call statics as instance methods just fine.

Comment (legoktm) [inline src/Semver.php]: Or just explicitly use the class name instead of "self".

Comment (Seldaek) [inline src/Semver.php]: Ah no you can call as instance if you have an instance, but it's in a static method already, so you just need to replace self:: with the class name I think

Comment (Seldaek) [inline src/Semver.php]: Damn beaten to it by 8seconds :p

**Pull request merged**